wordpress-mobile/WordPress-Android
# Issue/5530 basic notification channels android o

Fixes #5530

**Note:** Please note this PR only creates the needed Notification Channels for Android O, but does not care for the other new restrictions on the platform.([#](https://developer.android.com/about/versions/oreo/android-8.0-migration.html) and [#](https://developer.android.com/about/versions/oreo/background.html)) These will be addressed in other PRs.

This PR creates the Notification Channels as required in Android Oreo, and updates all of the instances of `NotificationCompat.Builder` constructor calls to pass the channel ID.

This PR creates 2 notification channels (named "Important" and "General"), because the 2fa auth notification should have a maximum priority and now the priorities cannot be set on the Notifications themselves but rather belong to the Channels these notifications belong to on Android O+ devices.

To test:
On Android 8+
1. Run the app
2. Now go to the device's Settings screen -> Apps & notifications -> WordPress
3. Tap on "App Notifications"
4. You should see the 2 new channels created under the section labelled "Categories":
a. Important
b. General

![screenshot_20180404-141901](https://user-images.githubusercontent.com/6597771/38323383-79ffed5e-3813-11e8-80ea-93d70b413b32.png)

On Android N and lower:
- Nothing should change.

cc @nbradbury
